### PR TITLE
[CELEBORN-1693] Fix storageFetcherPool concurrent problem

### PR DESCRIPTION
**Pull request by reswqa**
### What changes were proposed in this pull request?

Fix storageFetcherPool concurrent problem.

There may be duplicate thread pools created as multi-thread race condition.

image

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
No need.

### CODE REVIEW

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java [hunk 1]
```diff
@@ -44,7 +44,8 @@ public class CreditStreamManager {
   private final AtomicLong nextStreamId;
   private final ConcurrentHashMap<Long, StreamState> streams;
   private final ConcurrentHashMap<FileInfo, MapPartitionData> activeMapPartitions;
-  private final HashMap<String, ExecutorService> storageFetcherPool = new HashMap<>();
+  private final ConcurrentHashMap<String, ExecutorService> storageFetcherPool =
```
Comment (RexXiong) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java:47]: use JavaUtils.newConcurrentHashMap instead.

Comment (reswqa) [reply worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java:47]: Done

**Comment by FMX** [comment]
This surely improved the potential concurrent issue of the storage fetch pool. But are there any known issues related to this point?

**Comment by RexXiong** [comment]
> This surely improved the potential concurrent issue of the storage fetch pool. But are there any known issues related to this point?

There may be duplicate thread pools created, pls describe this @reswqa see:
image

**Comment by reswqa** [comment]
> > This surely improved the potential concurrent issue of the storage fetch pool. But are there any known issues related to this point?
> 
> There may be duplicate thread pools created, pls describe this @reswqa see: image

Thanks, update.

**Review by RexXiong — APPROVED** [review]

**Comment by RexXiong** [comment]
CC @SteNicholas

**Review by SteNicholas — APPROVED** [review]

**Comment by SteNicholas** [comment]
Merged to main(v0.6.0).

**Pull request closed without merging**